TypeCobol/Compiler/Diagnostics/CrossChecker.cs

```diff
@@ -755,6 +755,95 @@ public override bool Visit(DataDefinition dataDefinition)
+            {
+                if (dataEntry == null)
+                {
+                    dataEntry = dataDefinition.CodeElement as CommonDataDescriptionAndDataRedefines;
+                }
+
+                if (dataEntry?.Usage != null &&
+                    (dataEntry.Usage.Value == DataUsage.FloatingPoint || dataEntry.Usage.Value == DataUsage.LongFloatingPoint) &&
+                    dataEntry.Picture != null)
+                {
+                    DiagnosticUtils.AddError(dataDefinition,
+                        "Variable with usage COMP-1 and COMP-2 cannot have a PICTURE", dataEntry);
+                }
+
+                if (dataEntry?.LevelNumber != null && IsDataDefinitionEmpty(dataDefinition, dataEntry))
+                {
+                    //Get current node index
+                    var nodeIndex = dataDefinition.Parent.IndexOf(dataDefinition);
+                    //Get sibling nodes
+                    var siblingNodes = dataDefinition.Parent.Children;
+                    //Get immediately following DataDefinition
+                    var nextData = siblingNodes.Skip(nodeIndex + 1).OfType<DataDefinition>().FirstOrDefault();
+                    if (nextData != null && nextData.IsInsideCopy())
+                    {
+                        DiagnosticUtils.AddError(dataDefinition, $"Cannot include copy {nextData.CodeElement.FirstCopyDirective.TextName} " +
+                                                                  $"under level {dataEntry.LevelNumber} " +
+                                                                  $"because copy starts at level {nextData.CodeElement.LevelNumber}.", dataEntry);
+                    }
+                    else
+                    {
+                        DiagnosticUtils.AddError(dataDefinition, "A group item cannot be empty. Add children, picture or usage declaration.", dataEntry);
+                    }
+                }
+            }
+        }
+
+        private static bool IsDataDefinitionEmpty([NotNull] DataDefinition dataDefinition, [NotNull] CommonDataDescriptionAndDataRedefines dataEntry)
+        {
+            if (dataDefinition.IsTableOccurence)
+            {
+                // OCCURS: not empty if PICTURE or type definition
+                //TODO We systematically throw an error in case of empty array with a TYPEDEF defined on parent group
+                //This could be improved by checking after type expansion
+                if ((dataEntry.Picture != null) || (dataDefinition.TypeDefinition != null))
+                {
+                    return false;
+                }
+            }
+            else
+            {
+                // Others: exclude RENAME + not empty if PICTURE or user defined data type
+                if (dataDefinition.CodeElement.LevelNumber?.Value >= 50 || dataDefinition.Picture != null || dataEntry.UserDefinedDataType != null)
```

Comment:
What about level 77 ? Also you can use `dataEntry` instead of `dataDefinition.CodeElement`.

Reply:
Level >=50 cannot have children so they are always elementary item.
And :
- levels 66 and 88 don't have a picture/usage. 
- level 77 must have a picture/usage. But for me it's another separated test.

Reply:
Level 77 are now checked (+ specific tests in `Level77AndUsage.rdzb.cbl`)